### PR TITLE
#164829210 add feature to like or dislike an article

### PR DESCRIPTION
**Pull request by Sm0k3s**
#### Description
Users can like or dislike an article

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [x] End to end
- [x] Integration

#### Checklist:
- [ ] My code follows the style guide for this project
- [ ] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### How should this be manually tested?
1.  Clone the repository 
```https://github.com/andela/ah-centauri-backend.git```
2. On completing of cloning
```bash 
cd ah-centauri-backend
git checkout feature/164829209-user-can-comment-on-article
```
3. Rename the .env.example file into .env and update the parameters with your own keys.

> For Centauri team members you can use the keys shared through slack

4. Start development server and send the following through postman

**Like an article**
```

POST /api/articles/:slug/like/
Authentication required, likes an article and returns the total number of likes and dislikes for that article. 

```
**Dislike an article**
```

POST /api/articles/:slug/dislike/
Authentication required, dislikes an article and returns the total number of likes and dislikes for that article. 

```

#### Pivotal Tracker
[#164829210](https://www.pivotaltracker.com/n/projects/2321088/stories/164829210)

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -70,4 +74,33 @@ def delete(self, request, slug, format=None):
...
+    model = None    # Data Model - Articles or Comments
+    vote_type = None  # Vote type Like/Dislike
+    permission_classes = (IsAuthenticatedOrReadOnly, IsVerifiedUser, )
+
+    def post(self, request, slug):
+        obj = self.model.objects.get(slug=slug)
+        # GenericForeignKey does not support get_or_create
+        ct = ContentType.objects.get_for_model(obj)
+        try:
+            likedislike = LikeDislike.objects.get(
```
Comment (ValerieRono) [inline authors/apps/articles/views.py]: use snake_case to name your variables

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -70,4 +74,33 @@ def delete(self, request, slug, format=None):
         self.check_object_permissions(request, article)
         article.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
-    
+
+
+class LikesView(APIView):
+    model = None    # Data Model - Articles or Comments
```
Comment (ValerieRono) [inline authors/apps/articles/views.py:283]: use docstring to give info on what the class is about, and all other methods you write

File: authors/apps/articles/urls.py [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
 app_name = 'articles'
 
 urlpatterns = [
     path('articles/', CreateArticlesAPIView.as_view(), name='articles'),
-    path('article/<slug:slug>/', RetrieveUpdateDeleteArticleAPIView.as_view(), name='article')
-]
\ No newline at end of file
+    path('article/<slug:slug>/',
+         RetrieveUpdateDeleteArticleAPIView.as_view(), name='article'),
+    path('article/<slug:slug>/like/',
```
Comment (ValerieRono) [inline authors/apps/articles/urls.py]: change urls to follow pattern agreed on by the group

**Review by ValerieRono — COMMENTED** [review]
feature works, just a few changes requested with regards to coding conventions

File: authors/apps/articles/views.py (hunk 1 above)
Comment (Sm0k3s) [reply authors/apps/articles/views.py]: I have implemented this feedback

File: authors/apps/articles/views.py (hunk 2 above)
Comment (Sm0k3s) [reply authors/apps/articles/views.py:283]: I have added the docstrings

File: authors/apps/articles/urls.py (hunk 3 above)
Comment (Sm0k3s) [reply authors/apps/articles/urls.py]: I'll rebase for this

**Review by ValerieRono — APPROVED** [review]
looks good to me Obaga

**Review by ogiste — APPROVED** [review]
Added docstrings, make sure the functions/methods also have docstrings.

File: authors/apps/articles/views.py [hunk 4]
```diff
@@ -270,4 +275,38 @@ def delete(self, request, pk, format=None):
...
+    View for liking and disliking Articles and later on Comments
+    """
+    model = None    # Data Model - Articles or Comments
+    vote_type = None  # Vote type Like/Dislike
+    permission_classes = (IsAuthenticatedOrReadOnly, IsVerifiedUser, )
+
+    def post(self, request, *args, **kwargs):
+        obj = self.model.objects.get(slug=self.kwargs['slug'])
+        # GenericForeignKey does not support get_or_create
+        ct = ContentType.objects.get_for_model(obj)
```
Comment (oagutu) [inline authors/apps/articles/views.py]: Use a more descriptive variable name. Also add docstrings to this method

**Review by oagutu — COMMENTED** [review]
@Sm0k3s This is good. Just fix the one small feedback and let me know so I can merge it

File: authors/apps/articles/views.py (hunk 4 above)
Comment (Sm0k3s) [reply authors/apps/articles/views.py]: I have added the docstring and the descriptive variable name

**Pull request merged**